core/capabilities/webapi/round_robin_selector_test.go

```diff
@@ -10,7 +10,7 @@ import (
 
 func TestRoundRobinSelector(t *testing.T) {
```

Comment:
It would be good to add unit tests to show the new behavior in the PR.
I'd expect
- one showing how on start a random index is chosen
- one showing how fixed start always chooses index 0

Reply:
all the tests right now use `WithFixedStart` and these verify that this forces a start at 0 index.  There's not really a good way to assert the start index is random other than generate a distribution and make some assertions about it that can then be flaky.  Seems overkill.  Did you have something quicker/easier than asserting a distribution of start indexes in mind?

Reply:
It would require a small rework to use `rand` via dependency injection.
If not given default to `r:= rand.New(rand.NewSource(rand.Int63()))`
This lets you pass a seeded Rand in test for deterministic behavior.
https://pkg.go.dev/math/rand#Rand.Seed

Though this a nit because this is a pretty low stakes feature.